readthedocs/notifications/messages.py

```diff
@@ -29,8 +34,26 @@ def __repr__(self):
     def __str__(self):
         return f"Message: {self.id} | {self.header}"
 
+    def _escape_format_values(self, format_values):
+        """
+        Escape all potential HTML tags included in format values.
+
+        This is a protection against rendering potential values defined by the user.
+        It uses the Django's util function ``escape`` (similar to ``|escape`` template tag filter)
+        to convert HTML characters into regular characters.
+
+        NOTE: currently, we don't support values that are not ``str`` or ``int``.
+        If we want to support other types or nested dictionaries,
+        we will need to iterate recursively to apply the ``escape`` function.
+        """
+        return {
+            key: escape(value)
+            for key, value in format_values.items()
+            if isinstance(value, (str, int))
```

Comment:
I think this is fine without this check, there are more built-in types that can be included (like bool).

```suggestion
```

We should have more tests instead.

Reply:
Yeah, we can expand to other types in the future. I want to avoid running `escape(dict)` and returning 500. For now, `str` and `int` are the ones we are using.